config/http_config.go

```diff
@@ -785,33 +815,41 @@ func NewTLSRoundTripper(
 		return nil, err
 	}
 	t.rt = rt
-	_, t.hashCAFile, err = t.getCAWithHash()
+	_, t.hashCAFile, t.hashCertFile, t.hashKeyFile, err = t.getTLSFilesWithHash()
 	if err != nil {
 		return nil, err
 	}
 
 	return t, nil
 }
 
-func (t *tlsRoundTripper) getCAWithHash() ([]byte, []byte, error) {
-	b, err := readCAFile(t.caFile)
+func (t *tlsRoundTripper) getTLSFilesWithHash() ([]byte, []byte, []byte, []byte, error) {
+	b1, err := readCAFile(t.caFile)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, nil, err
+	}
+	h1 := sha256.Sum256(b1)
+
+	b2, b3, err := readCertAndKey(t.certFile, t.keyFile)
```

Comment:
Where do we deal with unset certFile and t.keyFile?

Reply:
Could you please elaborate on what you mean by unset?

Reply:
I mean empty strings

Reply:
Good point. A round tripper just isn't created if the `CAFile` is unset, but we can't do the same with the `certFile` and `keyFile`.

https://github.com/prometheus/common/blob/102c12a0a1cca70d642421c7cb151eaa701295bd/config/http_config.go#L452-L455

Reply:
Should be fixed.